signer/signer.go

```diff
@@ -16,7 +15,7 @@ type SigningService interface {
 
 // CryptoServiceIndex represents a mapping between a service algorithm string
 // and a CryptoService
-type CryptoServiceIndex map[data.KeyAlgorithm]signed.CryptoService
+type CryptoServiceIndex map[string]signed.CryptoService
```

Comment:
Everything looks cleaner until you get to these things.

`map[data.KeyAlgorithm]` is super easy to understand. `map[string]` you have to wonder what `string` is.